hive-plugins/src/main/java/co/cask/plugin/etl/batch/commons/HiveSchemaConverter.java

```diff
@@ -49,12 +49,14 @@
    */
   public static HCatSchema toHiveSchema(Schema schema, HCatSchema tableSchema) {
     List<HCatFieldSchema> fields = Lists.newArrayList();
+    //TODO: Finalize a consistent way to whether do projection on source or not while reading and if needed
```

Comment:
The inconsistent/unknown part is whether we should take schema(field and type) from the user or just the fields (like database plugin). After the above comment and @nitinmotgi  comment I am assuming we want to take the schema so I will add the type check part.